flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/reader/KafkaPartitionSplitReader.java

```diff
@@ -161,6 +171,13 @@ public KafkaPartitionSplitReader(
                                 finishedPartitions,
                                 recordsBySplits);
                     }
+                    // Update I/O metric
+                    sourceReaderMetricGroup
+                            .getIOMetricGroup()
+                            .getNumBytesInCounter()
+                            .inc(
```

Comment:
I have not much experience with how Kafka records are structured but did you consider also counting the metadata fields?

Reply:
Maybe it is better to directly forward the Kafka metric https://github.com/apache/flink/commit/a1441c262f35407c4029f6ddf8daa18cd859f822#diff-026b7f554eb085f1d2dde933dcafa4bf39abd995dcb071126a545c9d9de109fbR41